xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -2019,11 +2022,13 @@ where
         payload: &GroupUpdated,
         storage: &impl XmtpMlsStorageProvider,
     ) -> Result<bool, GroupMessageProcessingError> {
-        if self.dm_id.is_none() {
-            // Don't dedupe for regular groups.
+        if self.dm_id.is_none() || payload.added_inboxes.is_empty() {
```

Comment:
as you mentioned think this should be good performance wise because of this check here 👌